.github/workflows/build_release_binaries.yml

```diff
@@ -0,0 +1,88 @@
+name: Create Draft Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: Tag name for the release.
+        required: true
+
+jobs:
+  create_draft_release:
+    name: Create Draft Release
+    runs-on: ubuntu-latest
+    outputs:
+      upload_url: ${{ steps.create_release.outputs.upload_url }}
+    steps:
+      - name: Create release
+        id: create_release
+        uses: actions/create-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: ${{ github.event.inputs.version }}
+          release_name: Release v${{ github.event.inputs.version }}
+          draft: true
+
+  build_release_binary:
+    needs: create_draft_release
+    name: Build Release Binary
+    strategy:
+      matrix:
+        include:
+          - runner: ubuntu-latest
+            os: linux
+
+          - runner: windows-latest
+            os: windows
+
+          - runner: macos-latest
+            os: macos
+    runs-on: ${{ matrix.runner }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - uses: hashicorp/setup-terraform@v2
+        with:
+          terraform_wrapper: false
+
+      - name: Install Go toolchain
+        uses: actions/setup-go@v4
+        with:
+          go-version-file: 'src/go.mod'
+          cache-dependency-path: 'src/go.sum'
+
+      - name: Generate Binary Name
+        id: binary_name
+        shell: bash
+        run: |
+          binary_name="infra-tester"
+          release_binary_name="infra-tester-${{ matrix.os }}-x86_64"
+
+          if [[ ${{ matrix.os }} == "windows" ]]; then
+            binary_name="infra-tester.exe"
+            release_binary_name="infra-tester-${{ matrix.os }}-x86_64.exe"
```

Comment:
Nit: You can use 
```
binary_name="${binary_name}.exe"
release_binary_name="${release_binary_name}.exe"
```
to make it obvious that the only change is the appended `.exe` extension.